projects/lexical_test/lib/lexical/test/position_support.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Lexical.Test.PositionSupport do
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Line
+  alias Lexical.Document.Lines
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+
+  import Line
+
+  defmacro __using__(_) do
+    quote do
+      import unquote(__MODULE__)
+    end
+  end
+
+  @doc """
+  Normalizes any position in items on either side of the operation and runs
+  assert on the result.
+  """
+  defmacro assert_normalized({op, ctx, [left, right]}) do
+    normalized = {op, ctx, [do_normalize(left), do_normalize(right)]}
+
+    quote do
+      assert unquote(normalized)
+    end
+  end
+
+  def normalize(%Edit{} = edit) do
+    update_in(edit, [:range], &normalize/1)
+  end
+
+  def normalize(%Range{} = range) do
+    range
+    |> update_in([:start], &normalize/1)
+    |> update_in([:end], &normalize/1)
+  end
+
+  def normalize(%Position{} = position) do
+    %Position{
+      position
+      | context_line: nil,
```

Comment:
I actually don't quite understand the reason for normalization. Is it for more convenient testing?

Reply:
It’s so you can assert by making a position that doesn’t have all the fields set. Otherwise you’d need to copy the line, the offset, and other fields into the position, when all you care about is the line number and character.